### PR TITLE
New package: lilypond-2.19.65

### PR DESCRIPTION
**Pull request by newbluemoon**
This PR addresses issue #2987.

Lilypond is built without documentation. It will be added if/when the required package `dblatex` (#6347) becomes available. Though the packages `extractpdfmark` and `texi2html` are only used for documentation building I included them anyway. Lilypond upstream offers a pre-built documentation tarball which I used to create a `lilypond-doc` package (I think it’s good to have some offline doc for a program like lilypond).

Lilypond also depends on texlive. Instead of adding the `texlive-bin` package to `depends` (with all its implications) I added a post install message that says a working texlive installation is needed (a basic installation as done by `texlive-bin` should be sufficient).

A problem still is cross compilation: it builds natively without any problem (tested on x86_64, i686, and armv7l), but when cross compiling make just stops with a ‘no rule to make target’ error. Comparing the output of the nativ/cross configure scripts doesn’t show any difference. Right now I have no idea how to solve this, but thought I should open this PR anyway ;)

### CODE REVIEW

**Comment by newbluemoon** [comment]
I just noticed in the log: on Travis `automake` is installed but not `autoconf`

    ./autogen.sh: line 17: autoconf: command not found

But

    $ xbps-query -Rx automake
    perl>=0
    autoconf>=0

?

**Comment by chneukirchen** [comment]
t1utils are included in TeXLive.

I'm not very happy about adding yet another outdated unmaintained release of guile. Does this release still not support guile 2.0?

**Comment by newbluemoon** [comment]
Wasn’t aware that t1utils are in texlive. Will change that.

Guile 2 support is only mentioned as ‘highly experimental’. The last time I tried it didn’t work. But I can try again.

**Comment by newbluemoon** [comment]
Is there a way for a normal user to cancel a travis job? A previous build of lilypond (job no. 12568) has been running for over four hours now and is blocking others.

**Comment by newbluemoon** [comment]
Could you please cancel lilypond’s jobs on Travis no. 12571 and 12572? They have already been running for hours and all packages were built. But it seems that `show_files.sh` is inflating the log, because there are 27601 files in lilypond-doc and Travis somehow get stuck and does neither end nor abort.

**Comment by Vaelatern** [comment]
testing guile cross build.

**Comment by newbluemoon** [comment]
On x86_64 and i686 all packages were built, Travis seems to fail because the log gets too big (lilypond-doc contains more than 27000 files). The musl builds have to fail because there is no texlive.

**Comment by lemmi** [comment]
I can confirm the build works on `x86_64`, but fails on `x84_64-musl`. However `guile` seems to be missing as a dependency.
```
Log level set to 287
GNU LilyPond 2.19.65
Relocation: from PATH=/home/lemmi/bin:/home/lemmi/.gem/ruby/2.3.0/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin:/usr/share/apache-maven/bin:/opt/texlive/2017/bin/x86_64-linux
argv0=lilypond
PATH=/usr/bin (prepend)
Setting PATH to /usr/bin:/home/lemmi/bin:/home/lemmi/.gem/ruby/2.3.0/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin:/usr/share/apache-maven/bin:/opt/texlive/2017/bin/x86_64-linux
Relocation: compile datadir=, new datadir=/usr/share/lilypond//2.19.65
Relocation: framework_prefix=/usr/bin/..
Setting INSTALLER_PREFIX to /usr/bin/..
PATH=/usr/bin/../bin (prepend)
Setting PATH to /usr/bin/../bin:/usr/bin:/home/lemmi/bin:/home/lemmi/.gem/ruby/2.3.0/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin:/usr/share/apache-maven/bin:/opt/texlive/2017/bin/x86_64-linux
Setting GUILE_MIN_YIELD_1 to 65
Setting GUILE_MIN_YIELD_2 to 65
Setting GUILE_MIN_YIELD_MALLOC to 65
Setting GUILE_INIT_SEGMENT_SIZE_1 to 10485760
Setting GUILE_MAX_SEGMENT_SIZE to 104857600
Setting GUILE_AUTO_COMPILE to 0
Setting GUILE_WARN_DEPRECATED to detailed
Setting XDG_CACHE_HOME to /usr/share/lilypond/2.19.65
Throw without catch before boot:
Throw to key misc-error with args ("primitive-load-path" "Unable to find file ~S in load path" ("ice-9/boot-9") #f)Aborting.
Aborted
```
Installing `guile` manually solved the problem and simple examples produced working pdfs.

**Comment by newbluemoon** [comment]
Hmm, when I check my local build libguile got added as a dependency:

<pre>
$ xbps-query --repository=hostdir/binpkgs/lilypond -Rx lilypond
python>=0
ghostscript>=0
glibc>=2.25_1
libguile>=2.0.9_3
gc>=7.1_1
pango>=1.24.0_1
glib>=2.18.0_1
fontconfig>=2.6.0_1
freetype>=2.3.9_1
libstdc++>=4.4.0_1
libgcc>=4.4.0_1
</pre>

I also tested some examples using both guile2 and guile1.8 and the pdfs looked the same.

**Comment by lemmi** [comment]
The dependency `("ice-9/boot-9")` is found in `guile`, not `libguille`:
```
# xlocate ice-9/boot-9
guile-2.0.14_1	/usr/lib/guile/2.0/ccache/ice-9/boot-9.go
guile-2.0.14_1	/usr/share/guile/2.0/ice-9/boot-9.scm
```

**Comment by newbluemoon** [comment]
Ah, sorry. Had this fixed when using guile1.8 and reintroduced it when switching to guile2. But that makes me wonder: when using ‘libguile’ depends on files located in ‘guile’ shouldn’t ‘libguile’ depend on ‘guile’?

**Comment by newbluemoon** [comment]
Just another thought: Support of guile2 is regarded as ‘highly experimental’. It builds and pdfs are created without a problem for now but this might not be the case for all score sheets. Debian e.g. has guile-1.8 bundled with lilypond just for internal use. So I was thinking to do something like that and add a build option for an internal guile-1.8?

**Comment by lemmi** [comment]
Before this gets burried any deeper: Since we don't have guile-1.xx available, we might aswell try 2.0. If we do run into issues, we can still fix them.

**Comment by newbluemoon** [comment]
I tried to combine guile 1.8 and lilypond in one template, but whatever I tried was just an ugly hack. That’s just something xbps-src isn’t supposed to do, I guess. The only reasonable way I came up with (when wanting to use 1.8 internally) is separately making a guile1.8-devel package with the only purpose to use when building lilypond and copying the binary guile1.8, the libs and the *.scm files to a location only lilypond knows about. But this would also add guile 1.8 to the repo.

So to sum it up: lilypond works best with guile 1.8 but if that’s not available I’m with @lemmi to give 2.0 a try.

**Comment by chneukirchen** [comment]
Well, then package guile 1.8. As it is dead upstream, it probably won't be a maintenance burden. :P

**Comment by newbluemoon** [comment]
Ok, I made lilypond use guile-1.8 again ;-)

**Comment by lemmi** [comment]
Travis doesn't like it at all. The compilation output is gigantic. 
x86_64 built and worked.

**Comment by lemmi** [comment]
@newbluemoon have you found a way to clean up the builds and have you had a chance to try on a musl system?

**Comment by newbluemoon** [comment]
@lemmi I didn’t try it on a musl system because there’s currently no texlive available.
One thing that inflates the output is `common/travis/show_files.sh` because lilypond-doc contains about 27.000 files. Is there an option to disable `show_files.sh`? Or I could remove lilypond-doc for now.

**Comment by lemmi** [comment]
As for the build log: Don't bother then.
Otherwise, I can't cross compile and musl doesn't want to build aswell. At least for now we should set nocross. For musl the reason seems to be texlive. For other archs I don't understand the error.
Please set nocross. Then we'll merge and hopefully come up with some solutions later on.

**Comment by newbluemoon** [comment]
@lemmi: Done.
In the initial post, last paragraph, I mentioned cross compiling: It just stops for no apparent reason. Do you get a different error? I think that’s somehow due to lilypond’s off-the-wall building system, but right now have absolutely no clue.
